src/main/java/parser/Parser.java

```diff
@@ -0,0 +1,162 @@
+package parser;
+
+import exception.InvalidInputFormatException;
+
+import enumeration.CommandType;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.Objects;
+import java.util.Scanner;
+
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.temporal.ChronoUnit;
+import java.util.ArrayList;
+
+import task.Task;
+import task.Deadline;
+import task.Todo;
+import task.Event;
+
+/**
+ * Parser class that handles user inputs
+ */
+public class Parser {
+
+    public Parser(){
+
+    }
+
+
+
+    /**
+     * Reads lines of text from a file and returns them as an {@code ArrayList} of strings.
+     * Each line from the file is added to the list. If the file cannot be found, an error message
+     * is printed, and an empty {@code ArrayList} is returned.
+     *
+     * @param filepath the path to the file to be read
+     * @return an {@code ArrayList} containing the lines of text from the file
+     */
+    public static ArrayList<String> scan(String filepath) {
+        try {
+            Scanner sc = new Scanner(new File(filepath));
+            ArrayList<String> out = new ArrayList<>();
```

Comment:
`out` can be something more descriptive like `tasks` (because it is going to store tasks from the scanned file right?)